pygradle-plugin/src/test/groovy/com/linkedin/gradle/python/tasks/action/pip/PipWheelActionTest.groovy

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.tasks.action.pip
+
+import com.linkedin.gradle.python.exception.PipInstallException
+import com.linkedin.gradle.python.extension.PythonDetailsTestDouble
+import com.linkedin.gradle.python.extension.WheelExtension
+import com.linkedin.gradle.python.tasks.exec.ExternalExecTestDouble
+import com.linkedin.gradle.python.util.DefaultEnvironmentMerger
+import com.linkedin.gradle.python.util.PackageSettings
+import com.linkedin.gradle.python.wheel.EmptyWheelCache
+import org.gradle.process.ExecSpec
+import org.gradle.testfixtures.ProjectBuilder
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+import static com.linkedin.gradle.python.tasks.action.pip.PipActionHelpers.packageInGradleCache
+
+class PipWheelActionTest extends Specification {
+
+    @Rule
+    TemporaryFolder temporaryFolder
+
+    def "wheel install uses environment"() {
+        setup:
+        def override = [foo: ['CPPFLAGS': '-I/some/custom/path/include',
+                              'LDFLAGS' : '-L/some/custom/path/lib -Wl,-rpath,/some/custom/path/lib']
+        ]
+        def settings = new PipActionHelpers.EnvOverridePackageSettings(temporaryFolder, override)
+        def execSpec = Mock(ExecSpec)
+        def action = createPipWheelAction(settings, execSpec)
+
+        when:
+        action.buildWheel(packageInGradleCache("foo-1.0.0.tar.gz"), [])
+
+        then:
+        1 * execSpec.environment(['CPPFLAGS': '-I/some/custom/path/include', 'LDFLAGS': '-L/some/custom/path/lib -Wl,-rpath,/some/custom/path/lib'])
+    }
+
+    def 'wheel install uses global options'() {
+        Map<String, List<String>> override = ['setuptools': ['--global-option', '--dummy-global-option']]
+        def settings = new PipActionHelpers.GlobalOptionOverridePackageSettings(temporaryFolder, override)
+        def execSpec = Mock(ExecSpec)
+        def action = createPipWheelAction(settings, execSpec)
+
+        when:
+        action.buildWheel(packageInGradleCache("setuptools-1.0.0.tar.gz"), [])
+
+        then:
+        1 * execSpec.commandLine(_) >> { List<List<String>> args ->
+            println args
+            def it = args[0]
+            def idx = it.indexOf('--dummy-global-option')
```

Comment:
Can we search for `--global-option` first and confirm that the `--dummy-global-option` is behind it as we do for other kinds of options in the remaining tests below?